common/src/main/java/com/regnosys/rosetta/common/serialisation/xml/RosettaXmlMapper.java

```diff
@@ -0,0 +1,292 @@
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.Optional;
+
+import com.rosetta.util.serialisation.TypeXMLConfiguration;
+import org.apache.commons.io.IOUtils;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+/**
+ * Custom XmlMapper that extends the standard XmlMapper and overrides the base readValue method
+ * to delegate to the super class.
+ */
+public class RosettaXmlMapper extends XmlMapper {
+    private static final long serialVersionUID = 1L;
+    private final RosettaXMLConfiguration config;
+    private final ClassLoader classLoader;
+
+    public RosettaXmlMapper(RosettaXMLConfiguration config, ClassLoader classLoader) {
+        super((JacksonXmlModule) null);
+        this.config = config;
+        this.classLoader = classLoader;
+
+    }
+
+    private <T> T readClassValueInternal(String content) throws IOException, ParserConfigurationException, SAXException, ReadValueException {
+        String outermostElementName = getOutermostElementName(content);
+        Optional<Class<T>> valueTypeForElement = getValueTypeForElement(outermostElementName);
+        if (valueTypeForElement.isPresent()) {
+            return super.readValue(content, valueTypeForElement.get());
+        }
+        throw new ReadValueException("Unable to find value type for top level element");
+    }
+
+    private <T> T readJavaTypeValueInternal(String content) throws IOException, ParserConfigurationException, SAXException, ReadValueException {
+        String outermostElementName = getOutermostElementName(content);
+        Optional<Class<T>> valueTypeForElement = getValueTypeForElement(outermostElementName);
+        if (valueTypeForElement.isPresent()) {
+            return super.readValue(content, convertClassToJavaType(valueTypeForElement.get()));
+        }
+        throw new ReadValueException("Unable to find value type for top level element");
+    }
+
+    @Override
+    public <T> T readValue(String content, Class<T> valueType) throws JsonProcessingException {
+        try {
+            return readClassValueInternal(content);
+        } catch (ParserConfigurationException | IOException | SAXException | ReadValueException e) {
+            return super.readValue(content, valueType);
+        }
+    }
+
+    @Override
+    public <T> T readValue(String content, JavaType valueType) throws JsonProcessingException {
+        try {
+            return readJavaTypeValueInternal(content);
+        } catch (ParserConfigurationException | IOException | SAXException | ReadValueException e) {
+            return super.readValue(content, valueType);
+        }
+    }
+
+    @Override
+    public <T> T readValue(File src, Class<T> valueType) throws IOException {
```

Comment:
While reading these overridden methods, which seem to convert everything to strings, I wonder two things:
- is there not a common protected method to override? (looking at `_readValue` and `_readMapAndClose` here)
- if I'm not mistaken, Jackson doesn't load up whole files into memory, which is what you're doing here. Should we be concerned about performance/memory usage? E.g., when reading folders full of of expectation files.